src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/InternalSvDiscoverFromLocalAssemblyContigAlignmentsSpark.java

```diff
@@ -0,0 +1,230 @@
+
+    @Argument(doc = "file containing non-canonical chromosome names (e.g chrUn_KI270588v1) in the reference, human reference (hg19 or hg38) assumed when omitted",
+            shortName = "nonCanoChrFile",
+            fullName = "nonCanonicalChromosomeNamesFile", optional = true)
+    public String nonCanonicalChromosomeNamesFile;
+
+    @Argument(doc = "output directory", shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+              fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME)
+    private String outputDir;
+
+    @Argument(doc = "output SAM files", shortName = "wSAM",
+              fullName = "writeSAM", optional = true)
+    private boolean writeSAMFiles = false;
+
+    @Override
+    public boolean requiresReference() {
+        return true;
+    }
+
+    @Override
+    public boolean requiresReads() {
+        return true;
+    }
+
+    @Override
+    public ReadFilter makeReadFilter() {
+        return ReadFilterLibrary.MAPPED;
+    }
+
+    @Override
+    protected void runTool(final JavaSparkContext ctx) {
+
+        final JavaRDD<GATKRead> reads = getReads();
+        final SAMFileHeader header = getHeaderForReads();
+
+        // filter alignments and split the gaps
+        final JavaRDD<AlignedContig> contigsWithAlignmentsReconstructed =
+                InternalFilterLongReadAlignmentsSAMSpark.filterByScore(reads, header, nonCanonicalChromosomeNamesFile, localLogger)
+                        .filter(lr -> lr.alignmentIntervals.size()>1).cache();
+
+        // split up the long reads by their possible type of SV
+        final EnumMap<RawTypes, JavaRDD<AlignedContig>> splitUpLongReads =
+                splitReadsByPossiblyRawTypes(contigsWithAlignmentsReconstructed, localLogger);
+
+        if ( !FileUtils.createDirToWriteTo(outputDir) )
+            throw new GATKException("Could not create directory " + outputDir + " to write results to.");
+
+        if (writeSAMFiles) {
+            final Broadcast<SAMFileHeader> headerBroadcast = ctx.broadcast(header);
+            splitUpLongReads.forEach((k, v) -> writeSAM(v, k.name(), reads, headerBroadcast, outputDir, localLogger));
+        }
+
+        new InsDelVariantDetector()
+                .inferSvAndWriteVCF(splitUpLongReads.get(RawTypes.InsDel), outputDir+"/"+RawTypes.InsDel.name()+".vcf",
+                        ctx.broadcast(getReference()), discoverStageArgs.fastaReference,
+                        localLogger);
+    }
+
+    private enum RawTypes {
+        Ambiguous, Inv, InsDel, DispersedDupOrMEI, Cpx;
+    }
+
+    private static boolean hasOnly2Alignments(final AlignedContig contigWithOnlyOneConfig) {
+        return contigWithOnlyOneConfig.alignmentIntervals.size() == 2;
+    }
+
+    private static boolean isSameChromosomeMapping(final AlignedContig contigWithOnlyOneConfigAnd2Aln) {
+        Utils.validateArg(hasOnly2Alignments(contigWithOnlyOneConfigAnd2Aln),
```

Comment:
it seems like you could shorten these up if you made methods out of them, ie. `validateTwoAlignments()`, so you wouldn't have to repeat the logic and error message everywhere